engine/flink/util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/Aggregator.scala

```diff
@@ -0,0 +1,44 @@
+package pl.touk.nussknacker.engine.flink.util.transformer.aggregate
+
+import cats.data.{NonEmptyList, Validated, ValidatedNel}
+import org.apache.flink.api.common.functions.AggregateFunction
+import pl.touk.nussknacker.engine.api.LazyParameter
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{CannotCreateObjectError, NodeId}
+import pl.touk.nussknacker.engine.api.context.{ContextTransformationDef, ProcessCompilationError, ValidationContext}
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+
+/*
+  This class serves two purposes:
+  - computeOutputType provides information for validation, completion etc.
+  - we provide own versions of AggregateFunction methods to use type members instead of type parameters, as it makes it easier to use
+ */
+trait Aggregator extends AggregateFunction[Any, Any, Any] {
+
+  type Aggregate <: AnyRef
+
+  type Element <: AnyRef
+
+  def zero: Aggregate
+
+  def add: (Element, Aggregate) => Aggregate
```

Comment:
Why this method, `merge` and `result` return function instead of simple handle `Element` and `Aggregate` as a method's parameters?

Reply:
true, I changed it